xocto/fields/postgres/ranges.py

```diff
@@ -158,22 +166,18 @@ class HalfFiniteDateTimeRangeField(pg_fields.DateTimeRangeField):
     """
 
     def get_prep_value(
-        self, value: Optional[ranges.HalfFiniteDatetimeRange]
-    ) -> Optional[pg_ranges.DateTimeTZRange]:
+        self, value: Any
```

Comment:
Why `Any` here and not `Optional[Union[ranges.HalfFiniteDatetimeRange, datetime.datetime]]`? (And similar for the other `get_prep_value` implementations).

Reply:
By my understanding, `Any` will be a more correct type hint here. We can't enforce that the value will be `Optional[Union[ranges.HalfFiniteDatetimeRange, datetime.datetime]]`, since there is nothing stopping us doing things like `MyModel.objects.filter(period__contains="foo")` (or any other arbitrary type). This type checking happens at runtime (hence the `raise TypeError`), not mypy.